src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -660,12 +661,13 @@ export const Deposit: React.FC<DepositProps> = ({
       }
 
       const isSweepNeededQueryKey: IsSweepNeededQueryKey = ['isSweepNeeded', isSweepNeededQueryArgs]
-      const _isSweepNeeded = isSweepNeededQueryEnabled
-        ? await queryClient.fetchQuery({
-            queryKey: isSweepNeededQueryKey,
-            queryFn: isSweepNeededQueryFn,
-          })
-        : undefined
+      const _isSweepNeeded =
+        isSweepNeededQueryEnabled && isGetSweepNeededInput(isSweepNeededQueryArgs)
```

Comment:
We didn't have type-safety here and the `queryFn` itself was handling this by throwing, which is wrong - now decoupled from react-query by using `getIsSwepNeeded` async fn directly, which is not accepting partials